tests/lib/c_lib/src/main.c

```diff
@@ -319,6 +322,181 @@ void test_bsearch(void)
 	zassert_not_null(result, "bsearch -key found");
 }
 
+/**
+ *
+ * @brief Test abs function
+ *
+ */
+
+void test_abs(void)
+{
+	int val = -5, value = 5;
+
+	zassert_equal(abs(val), 5, "abs -5");
+	zassert_equal(abs(value), 5, "abs 5");
+}
+
+/**
+ *
+ * @brief Test atoi function
+ *
+ */
+
+void test_atoi(void)
+{
+	char *str = "123";
+
+	zassert_equal(atoi(str), 123, "atoi error");
+}
+
+/**
+ *
+ * @brief Test value type
+ *
+ * @details This function check the char type,
+ * and verify the return value.
+ *
+ * @see isalnum(), isalpha(), isdigit(), isgraph(),
+ * isprint(), isspace(), isupper(), isxdigit().
+ *
+ */
+void test_checktype(void)
+{
+	int str1 = '1',  str2 = 'a', str3 = '!', str4 = ' ', str5 = 'z';
+	int ch = 0x7f, upper = 'A';
+
+	zassert_true(isalnum(str1), "isalnum '1'");
+	zassert_true(isalnum(str2), "isalnum 'a'");
+	zassert_false(isalnum(str3), "isalnum '!'");
+
+	zassert_false(isalpha(str1), "isalpha '1'");
+	zassert_true(isalpha(str2), "isalpha 'a'");
+
+	zassert_true(isdigit(str1), "isdigit '1'");
+	zassert_false(isdigit(str2), "isdigit 'a'");
+
+	zassert_true(isgraph(str1), "isgraph 1");
+	zassert_false(isgraph(str4), "isgraph ' '");
+
+	zassert_false(isprint(ch), "isprint 0x7f");
+	zassert_true(isprint(str1), "isprint '1'");
+
+	zassert_false(isspace(str2), "isspace 'a'");
+	zassert_true(isspace(str4), "isspace ' '");
+
+	zassert_false(isupper(str2), "isupper 'a'");
+	zassert_true(isupper(upper), "isupper 'A'");
+
+	zassert_true(isxdigit(str2), "isxdigit 'a'");
+	zassert_false(isxdigit(str5), "isxdigit 'z'");
+}
+
+/**
+ *
+ * @brief Test character operation
+ *
+ * @see memchr(), memcpy(), memmove().
+ *
+ */
+void test_memstr(void)
+{
+	const char str[] = "testfunction";
+	int ch1 = 'e', ch2 = 'z';
+	char arr[10], move_arr[6] = "12123";
+	const char num[6] = "12121";
+
+	zassert_not_null(memchr(str, ch1, strlen(str)),
+		     "memchr 'testfunction' 'e' ");
+	zassert_is_null(memchr(str, ch2, strlen(str)),
+		     "memchr 'testfunction' 'z'");
+
+	zassert_not_null(memcpy(arr, num, sizeof(num)), "memcpy return NULL");
+	zassert_equal(memcmp(num, arr, sizeof(num)), 0,
+		     "memcpy failed");
+
+	zassert_not_equal(memmove(move_arr + 2, move_arr, 3), NULL,
```

Comment:
I check `memmove()` return if it's NULL first, after this assert, I have check the result already.

Reply:
You have not verified that `memmove()` returns the value of its first argument.  You're only checking whether it returns a non-null pointer.  Please fix it.

You should also check the exact return value of the other two functions.

Reply:
I think `move_arr` is `memmove()` return value. So I check `move_arr` whether correct at `next rows` like this `zassert_equal(memcmp(num, move_arr, 6), 0, "memmove failed");`.  Do you think it's right?

Reply:
No.
memmove returns the pointer it was given as the destination, which above is `move_arr + 2`.  So you want:
```
zassert_equal(memmove(move_arr + 2, move_arr, 3), move_arr + 2,
```
Similarly make sure that `memcpy` and `memchr` return the correct non-null pointer value.  `memcpy` returns an integer, so that you would check whether its negative, zero, or positive.